dspace-swordv2/src/main/java/org/dspace/sword2/WorkflowTools.java

```diff
@@ -94,10 +116,15 @@ public InProgressSubmission getWorkflowItem(Context context, Item item)
     {
         try
         {
-            if(ConfigurationManager.getProperty("workflow","workflow.framework").equals("xmlworkflow")){
-                return XmlWorkflowItem.findByItem(context, item);
-            }else{
-                return BasicWorkflowItem.findByItem(context, item);
+            if (ConfigurationManager
```

Comment:
Perhaps replace this by: WorkflowServiceFactory.getInstance().getWorkflowService().findByItem(context, item) ? This will work for both workflows at once & leave out the need for the old if this then that.